src/smtsolvers/TheoryIF.cc

```diff
@@ -88,16 +89,18 @@ TPropRes CoreSMTSolver::handleNewSplitClauses(SplitClauses & splitClauses) {
         auto & splitClause = splitClauses[index];
         unsigned satisfied = 0;
         unsigned unknown = 0;
+        std::optional<int> notFalsifiedIndex = std::nullopt;
```

Comment:
Making `impliedIndex` optional wasn't necessary tbh, but I see why you did it...
Overall PR is good!

Reply:
True. But I prefer to be explicit in the code in saying "Here I don't have any value yet!" and "Here I have a value!"